File: Javascript/listaenlazada.js
class Nodo {
    constructor(valor) {
        this.valor = valor;
        this.siguiente = null;
    }
}

class ListaEnlazada {
    constructor() {
        this.cabeza = null;
    }

    // Insertar un valor al final de la lista
    insertar(valor) {
        const nuevoNodo = new Nodo(valor);
        if (!this.cabeza) {
            this.cabeza = nuevoNodo;
        } else {
            let actual = this.cabeza;
            while (actual.siguiente) {
                actual = actual.siguiente;
            }
            actual.siguiente = nuevoNodo;
        }
    }

    // Mostrar la lista
    mostrar() {
        let actual = this.cabeza;
        while (actual) {
            console.log(actual.valor, end=" ");
            actual = actual.siguiente;
        }
        console.log();
    }
}

// Crear una lista enlazada
const lista = new ListaEnlazada();

// Insertar valores en la lista
lista.insertar(1);
lista.insertar(2);
lista.insertar(3);

// Mostrar la lista
console.log("Lista:", end=" ");
lista.mostrar();

File: C/listaenlazada.c
#include <iostream>
using namespace std;

struct Nodo {
    int valor;
    Nodo* siguiente;
    
    Nodo(int val) : valor(val), siguiente(nullptr) {}
};

class ListaEnlazada {
public:
    ListaEnlazada() : cabeza(nullptr) {}

    // Insertar un valor al final de la lista
    void insertar(int valor) {
        Nodo* nuevoNodo = new Nodo(valor);
        if (cabeza == nullptr) {
            cabeza = nuevoNodo;
        } else {
            Nodo* actual = cabeza;
            while (actual->siguiente != nullptr) {
                actual = actual->siguiente;
            }
            actual->siguiente = nuevoNodo;
        }
    }

    // Mostrar la lista
    void mostrar() {
        Nodo* actual = cabeza;
        while (actual != nullptr) {
            cout << actual->valor << " ";
            actual = actual->siguiente;
        }
        cout << endl;
    }

private:
    Nodo* cabeza;
};

int main() {
    ListaEnlazada lista;

    // Insertar valores en la lista
    lista.insertar(1);
    lista.insertar(2);
    lista.insertar(3);

    // Mostrar la lista
    cout << "Lista: ";
    lista.mostrar();

    return 0;
}
